tests/Bridge/Laravel/Http/Middlewares/AuditMiddlewareTest.php

```diff
@@ -0,0 +1,288 @@
+<?php
+declare(strict_types=1);
+
+namespace Tests\LoyaltyCorp\Auditing\Bridge\Laravel\Http\Middlewares;
+
+use EoneoPay\Externals\Logger\Logger;
+use EoneoPay\Utils\DateTime;
+use Illuminate\Http\Request;
+use LoyaltyCorp\Auditing\Bridge\Laravel\Http\Middlewares\AuditMiddleware;
+use LoyaltyCorp\Auditing\Bridge\Laravel\Services\Interfaces\HttpLoggerInterface;
+use LoyaltyCorp\Auditing\Exceptions\InvalidDocumentClassException;
+use LoyaltyCorp\Auditing\Services\Factories\Psr7Factory;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Tests\LoyaltyCorp\Auditing\Stubs\Bridge\Laravel\Services\HttpLoggerExceptionStub;
+use Tests\LoyaltyCorp\Auditing\Stubs\Bridge\Laravel\Services\HttpLoggerStub;
+use Tests\LoyaltyCorp\Auditing\Stubs\Vendor\Monolog\Handler\LogHandlerStub;
+use Tests\LoyaltyCorp\Auditing\Stubs\Vendor\Symfony\HttpFoundation\ResponseStub;
+use Tests\LoyaltyCorp\Auditing\TestCase;
+use Zend\Diactoros\Response\TextResponse;
+
+/**
+ * @covers \LoyaltyCorp\Auditing\Bridge\Laravel\Http\Middlewares\AuditMiddleware
+ */
+class AuditMiddlewareTest extends TestCase
+{
+    /**
+     * Test basic handle function
+     *
+     * @return void
+     *
+     * @throws \EoneoPay\Utils\Exceptions\InvalidDateTimeStringException
+     */
+    public function testHandle(): void
+    {
+        $middleware = $this->getMiddleware();
+
+        $response = $middleware->handle(
+            $this->getRequest(),
+            static function (): Response {
+                return new Response('OK');
+            }
+        );
+
+        self::assertSame('OK', $response->getContent());
+    }
+
+    /**
+     * Test if handle can catch the exceptions and continue processing
+     * and rethrow exception at end
+     *
+     * @return void
+     *
+     * @throws \EoneoPay\Utils\Exceptions\InvalidDateTimeStringException
+     */
+    public function testHandleHandlesExceptionsAndRethrows(): void
+    {
+        $this->expectException(InvalidDocumentClassException::class);
+        $this->expectExceptionMessage('Something failed.');
+
+        $middleware = $this->getMiddleware();
+
+        $middleware->handle(
+            $this->getRequest(),
+            static function (): void {
+                throw new InvalidDocumentClassException('Something failed.');
+            }
+        );
+    }
+
+    /**
+     * Test handling of a response which is not instance of Response or ResponseInterface
+     *
+     * @return void
+     *
+     * @throws \EoneoPay\Utils\Exceptions\InvalidDateTimeStringException
+     */
+    public function testHandleResponseWhenItsNotKnownType(): void
```

Comment:
this test here is to check if the response from `$next($request)` is not of know type, its handled as being passed as a null to HttpLogger